src/EditorFeatures/CSharpTest/AddRequiredParentheses/AddRequiredParenthesesTests.cs

```diff
@@ -0,0 +1,158 @@
+                this.SingleOption(CodeStyleOptions.LogicalOperationParentheses, new CodeStyleOption<LogicalOperationParenthesesPreference>(LogicalOperationParenthesesPreference.RequireForPrecedenceClarity, NotificationOption.Suggestion)));
+
+        private IDictionary<OptionKey, object> ArithmeticRequireForClarity
+            => OptionsSet(
+                this.SingleOption(CodeStyleOptions.ArithmeticOperationParentheses, new CodeStyleOption<ArithmeticOperationParenthesesPreference>(ArithmeticOperationParenthesesPreference.RequireForPrecedenceClarity, NotificationOption.Suggestion)),
+                this.SingleOption(CodeStyleOptions.LogicalOperationParentheses, new CodeStyleOption<LogicalOperationParenthesesPreference>(LogicalOperationParenthesesPreference.RemoveIfUnnecessary, NotificationOption.Suggestion)));
+
+        private IDictionary<OptionKey, object> LogicalRequireForClarity
+            => OptionsSet(
+                this.SingleOption(CodeStyleOptions.ArithmeticOperationParentheses, new CodeStyleOption<ArithmeticOperationParenthesesPreference>(ArithmeticOperationParenthesesPreference.RemoveIfUnnecessary, NotificationOption.Suggestion)),
+                this.SingleOption(CodeStyleOptions.LogicalOperationParentheses, new CodeStyleOption<LogicalOperationParenthesesPreference>(LogicalOperationParenthesesPreference.RequireForPrecedenceClarity, NotificationOption.Suggestion)));
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddRequiredParentheses)]
+        public async Task TestArithmeticPrecedence()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        int x = 1 + 2 [||]* 3;
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        int x = 1 + (2 * 3);
+    }
+}", parameters: new TestParameters(options: AllRequireForClarity));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddRequiredParentheses)]
+        public async Task TestNoArithmeticOnLowerPrecedence()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        int x = 1 [||]+ 2 * 3;
+    }
+}", parameters: new TestParameters(options: AllRequireForClarity));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddRequiredParentheses)]
+        public async Task TestNotIfArithmeticPrecedenceStaysTheSame()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        int x = 1 + 2 [||]+ 3;
+    }
+}", parameters: new TestParameters(options: AllRequireForClarity));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddRequiredParentheses)]
+        public async Task TestNotIfArithmeticPrecedenceIsOff()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        int x = 1 + 2 [||]+ 3;
```

Comment:
I'm keeping this test (just for exhaustiveness), but i'm adding another test that actually verifies that we don't add teh parens where we normally would if hte optoin was on.